src/render/opengl/SDL_render_gl.c

```diff
@@ -1961,11 +1994,15 @@ SDL_RenderDriver GL_RenderDriver = {
     GL_CreateRenderer,
     { "opengl",
       (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC | SDL_RENDERER_TARGETTEXTURE),
-      4,
-      { SDL_PIXELFORMAT_ARGB8888,
-        SDL_PIXELFORMAT_ABGR8888,
-        SDL_PIXELFORMAT_RGB888,
-        SDL_PIXELFORMAT_BGR888 },
+      8,
+      { SDL_PIXELFORMAT_RGBA32,
+        SDL_PIXELFORMAT_RGBX32,
+        SDL_PIXELFORMAT_RGB24,
```

Comment:
This will likely not use less memory in practice (on desktop GL and in GLES) because GPUs don't like formats that have non power of two pixel byte sizes, so any time support is required by a graphics API drivers tend to use RGBA under the hood.